core/sodasql/cli/ingest.py

```diff
@@ -0,0 +1,189 @@
+"""
+CLI commands to ingest test results from various sources into the Soda cloud.
+"""
+
+from __future__ import annotations
+
+import datetime as dt
+import json
+import logging
+import os
+from pathlib import Path
+from typing import Iterator
+
+from sodasql.__version__ import SODA_SQL_VERSION
+from sodasql.scan.parser import Parser
+from sodasql.scan.scan_builder import (
+    build_warehouse_yml_parser,
+    create_soda_server_client,
+)
+from sodasql.scan.test import Test
+from sodasql.scan.test_result import TestResult
+
+
+
+def create_dbt_run_result_to_test_result_mapping(
+    test_nodes: dict[str, CompiledSchemaTestNode],
+    run_results: list[RunResultOutput],
+):
+    """
+    Create a mapping with which test result belong to which run result.
```

Comment:
I think this rephrasing might be easier to read:

Map test results to run result